tests/05synapse.pl

```diff
@@ -28,11 +28,15 @@ sub extract_extra_args
 # see
 #   https://rt.perl.org/Public/Bug/Display.html?id=128774
 main::AT_END sub {
-   $OUTPUT->diag( "Killing synapse servers " ) if @synapses;
 
    ( fmap_void {
       my $synapse = $_;
 
+      # skip this if the process never got started.
+      return Future->done unless $synapse->{proc};
```

Comment:
That's a little bit of gut-wrenching into the innards of the `$synapse` object there. I'd use the `->pid` accessor method that you've already fixed to return zero if there isn't a process:

    return Future->done unless $synapse->pid;